### ISSUE TITLE
Getting Started page Install Dependencies step doesn't work (NNBD)

### ISSUE THREAD

**Issue by Tywholland** [mobxjs/mobx.dart#610]
I'm setting up a new Flutter project with the dart SDK version set to `sdk: ">=2.12.0-0 <3.0.0"` to enable null safety. When following the `Install Dependencies` section of https://mobx.netlify.app/getting-started/, the following error occurs when running `flutter packages get`:

```
packages get

Because mobx_codegen >=1.0.3 depends on mobx ^1.1.0 and app_namedepends on mobx ^2.0.0-nullsafety.2, mobx_codegen >=1.0.3 is forbidden.
So, because app_name depends on mobx_codegen ^1.1.2, version solving failed.

Running "flutter pub get" in AppName...

pub get failed (1; So, because app_name depends on mobx_codegen ^1.1.2, version solving failed.)
```

Could the Getting Started page be updated to resolve this issue? Or is this mostly just waiting for mobx_codegen to be updated?

### Notes:
The pubspec.yaml for this app looks like this:
```
environment:
  sdk: ">=2.12.0-0 <3.0.0"
  flutter: ">=1.17.0"

dependencies:
  flutter:
    sdk: flutter
  mobx: ^2.0.0-nullsafety.2
  flutter_mobx: ^2.0.0-nullsafety.0

dev_dependencies:
  flutter_test:
    sdk: flutter
  integration_test:
    sdk: flutter
  build_runner: ^1.10.9
  mobx_codegen: ^1.1.2
  pedantic: ^1.10.0-nullsafety.3

```

**Comment by KAROTT7** [comment]
I have same problem!!!!

**Comment by DavidTheProgrammer** [comment]
Until they update `mobx_codegen` with nullsafety I was able to get this running by downgrading to the latest *non-nullsafe* version of `mobx` & `flutter_mobx`

```
mobx: ^1.2.1+4
flutter_mobx: ^1.1.0+2

```

**Comment by pavanpodila** [comment]
Fixed with #623
